### PR TITLE
Custom cancelable implementation replaced with context

### PR DESCRIPTION
**Pull request by soffokl**

### CODE REVIEW

**Comment by vkuznecovas** [comment]
Mixed feelings here. 

On one hand, we get rid of the custom implementation of cancellable and resolve to using a standard tool for that. 

On the other hand, I feel like we end up passing a whole lot of contexts around for little benefit.

File: core/connection/manager.go [hunk 1]
```diff
@@ -89,76 +90,57 @@ func (manager *connectionManager) Connect(consumerID, providerID identity.Identi
...
-	val, err := cancelable.
-		NewRequest(func() (interface{}, error) {
-			return manager.findProposalByProviderID(providerID)
-		}).
-		Call()
+	var cancel []func()
+	defer func() {
+		manager.cleanConnection = func() {
+			manager.status = statusDisconnecting()
+			cancelCtx()
```
[the remark this answers is not in the document]
Comment (soffokl) [reply core/connection/manager.go]: I have tried this way, but calling `manager.cleanConnection` from `manager.cleanConnection` leads to the infinite recursion.

**Comment by soffokl** [comment]
OK, clear. 
I'll do it with a smaller spreading. 
And will ask for another review.
Thank you all for the feedback.

File: core/connection/manager.go [hunk 2]
```diff
@@ -47,16 +47,16 @@ var (
 )
 
 type connectionManager struct {
+	ctx context.Context
```
Comment (Waldz) [inline core/connection/manager.go]: Should be in second group of attributes

**Comment by soffokl** [comment]
@Waldz @vkuznecovas @zolia @tadovas please take a look on this once again. We, probably, should decide do we need it or not.

**Pull request merged**